### ISSUE TITLE
Handling non-rendered fields or parts in the admin in edit or create mode.

### ISSUE THREAD

**Issue by giannik** [OrchardCMS/OrchardCore#14551]
There are cases when you are creating or editing a content item that you do not want to render in the in the admin.
For example, I have some secret fields in my content type and I've created various alternate templates for specific roles so that these secret fields are not rendered.
You can also achieve the same thing with placement rules that do not render fields or parts based on custom logic.
**The problem is that when  update is performed for the content item the field is still updated even though it is not rendered, overriding the existing values in the secret fields.**

One approach to solve this is to add for all fields and parts an IsRendered  property like a flag on the viewmodel to avoid the update call in the driver: 

example code for textfield : 

```
//viewmodel 
 public class EditTextFieldViewModel
    {
      public bool IsRendered { get; set; }
    }		



//view
\OrchardCore.Modules\OrchardCore.ContentFields\Views\TextField.Edit.cshtml
<input type="hidden" asp-for="IsRendered" value="true"/>

texfield driver
  public override async Task<IDisplayResult> UpdateAsync(TextField field, IUpdateModel updater, UpdateFieldEditorContext context)
        {
            var viewModel = new EditTextFieldViewModel();
            if (await updater.TryUpdateModelAsync(viewModel, Prefix, f => f.IsRendered))
            {
                if (!viewModel.IsRendered) {
                      return Edit(field, context);
                }
//normal update 
}
```

While this works it requires changing the code for every field and part.

I'm wondering if there may be a better approach

If the community is willing I can submit a PR with this suggestion for all the known contentfields and parts

**Comment by hishamco** [comment]
> The problem is that when update is performed for the content item the field is still updated even though it is not rendered, overriding the existing values in the secret fields.

Good point, what about adding a property in the `ContentField` and doing the check in one place?

@MikeAlhayek do you have a better idea

**Comment by MikeAlhayek** [comment]
@giannik 
Why are you using content-fields for this? You should create a custom content Part via code and add a custom content part driver that does the logic you want.

public class CustomSecretPart : ContentPart 
{
    public string SecretThing { get; set; }
}

Then in the display driver on updating, you can securely set the secret for the role.

**Comment by hishamco** [comment]
He doesn't want to show the field nor updating its value while updating (when using placement)

**Comment by MikeAlhayek** [comment]
@hishamco 
Placement is just a virtual thing. Fields will still have to be validated, updated and saved. Content fields are not good for this custom scenario, Instead he should move all the fields that that are used for the secret things into custom part so he can control their behavior virtually and functionally. he can use the display driver to generate values, show anything he wants (if any) and do the custom logic he is after.

**Comment by hishamco** [comment]
Got it, can we move this to discussion, I think the answer credit should goes to Mike :)
